### DISCUSSION TITLE
[@types/styled-components] ThemeProvider cannot be used as a JSX component

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [DefinitelyTyped/DefinitelyTyped#59783] [Issues with a @types package] [answered]
So I think there is an issue with either the new @types/react@18 or @types/styled-components needs updating to support react 18.

Its currently throwing this issue saying that ThemeProvider cannot be used as a JSX component, but like how else are you supposed to use it, as its a context provider meaning it should be a JSX element. I'm assuming this is just React-18 teething issues?

Any help would be great.

I am also getting this error when doing 

const theme = useContext(ThemeContext);

```
Argument of type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").Context<any>' is not assignable to parameter of type 'React.Context<any>'.   
The types of 'Provider.propTypes' are incompatible between these types.     
Type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").WeakValidationMap<import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").ProviderProps<any>> | undefined' is not assignable to type 'React.WeakValidationMap<React.ProviderProps<any>> | undefined'.       
Type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").WeakValidationMap<import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").ProviderProps<any>>' is not assignable to type 'React.WeakValidationMap<React.ProviderProps<any>>'.         
Types of property 'children' are incompatible.           
Type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").Validator<import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").ReactNode> | undefined' is not assignable to type 'React.Validator<React.ReactNode> | undefined
Type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").Validator<import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").ReactNode>' is not assignable to type 'import("/Users/iv0706/code/TechStack/node_modules/@types/prop-types/index").Validator<React.ReactNode>'.
Type 'import("/Users/iv0706/code/TechStack/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.                 
Type '{}' is not assignable to type 'ReactNode'.
```

[a comment by an automated account is not shown]

**Reply by eps1lon** [reply]
Could you check if you have multiple versions of types/react installed? See https://github.com/react-native-svg/react-native-svg/issues/1741 for more info

**Reply by The-Code-Monkey** [reply]
In my yarn.lock there is @types/react@18.0.0 and @types/react@* which is pulling version "17.0.43" could that be causing the issue. its quite a big mono repo so there are probably sub libs pulling in the older versions. I guess i just need to add a resolution into my package json for @types/react@18

**Reply by The-Code-Monkey** [reply]
Hmm yeah that was the correct thing to do, i had to delete my yarn.lock and run yarn. which is strange because yarn is supposed to dedupe the lock file automatically.

**Reply by eps1lon** [reply]
Package managers do not dedupe when you add a dependency. Only when you specifically ask them to dedupe an existing lockfile

**Reply by The-Code-Monkey** [reply]
https://classic.yarnpkg.com/en/docs/cli/dedupe
